test/framework/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java

```diff
@@ -113,15 +112,13 @@ public static void assertConsistency(BlobStoreRepository repository, Executor ex
                 assertIndexUUIDs(repository, repositoryData);
                 assertSnapshotUUIDs(repository, repositoryData);
                 assertShardIndexGenerations(blobContainer, repositoryData.shardGenerations());
-                return null;
             } catch (AssertionError e) {
```

Comment:
I believe we're catching the `AssertionError` here in order to re-throw it on the calling thread so that we can use this within `assertBusy()` for the third-party tests that allow for S3 to be eventually-consistent. Do we need that any more, now that S3 is properly consistent?

Reply:
Good question ... I guess you could make the same argument for other eventual consistency code (mainly `EventuallyConsistentMockRepository` and just drop that stuff as well. But I always figured that we may still have other S3 compatible implementations  out there in the wild that are eventually consistent and so it's nice to have this stuff in tests for third party testing? (not sure ... I don't know of any and would also be happy to drop all of it to save some LoC :))

Reply:
I think we could reasonably say a third (fourth?) party implementation that has weaker consistency than the real S3 is not compatible. The repo analysis API fails on inconsistent listings:

https://github.com/elastic/elasticsearch/blob/68817d7ca29c264b3ea3f766737d81e2ebb4028c/x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalyzeAction.java#L580-L585

Reply:
fair point, I'll opera follow-up that drops all these tests in a bit then :)